src/dcd/server/autocomplete/calltip_utils.d

```diff
@@ -0,0 +1,40 @@
+module dcd.server.autocomplete.calltip_utils;
+import std.string;
+import std.regex;
+import std.range : empty;
+import std.experimental.logger;
+import std.algorithm : canFind;
+
+string removeFirstArgumentOfFunction(string callTip)
+{
+    auto parentheseSplit = callTip.split('(');
+    // has only one argument
+    if (!callTip.canFind(','))
+    {
+        return parentheseSplit[0] ~ "()";
+    }
+    auto commaSplit = parentheseSplit[1].split(',');
+    string newCallTip = callTip.replace((commaSplit[0] ~ ", "), "");
+    return newCallTip;
+
+}
+
+unittest
+{
+    auto result = removeFirstArgumentOfFunction("void fooFunction(const(immutable(Foo)) foo)");
+    assert(result, "void fooFunction()");
+}
+
+unittest
+{
+    auto result = removeFirstArgumentOfFunction(
+        "void fooFunction(const(immutable(Foo)) foo), string message");
```

Comment:
you mean we don't need to remove the first parameter right?

Reply:
with letting the IDE do the job I meant offsetting the "current parameter" by one, which IDEs do already anyway, if they show the currently active parameter. I think it's useful to have the first parameter type still, especially because this is still not 100% accurate and the user might catch issues with it that way.

Reply:
alright let's just keep the first parameter as is.